rpc/internal/buildlogger_service.go

```diff
@@ -66,6 +67,24 @@ func (s *buildloggerService) StreamLog(stream Buildlogger_StreamLogServer) error
 	return nil
 }
 
+// CloseLog "closes out" a buildlogger log by setting the completed at
+// timestamp and the exit code. This should be the last rcp call made on a log.
 func (s *buildloggerService) CloseLog(ctx context.Context, info *LogEndInfo) (*BuildloggerResponse, error) {
-	return nil, nil
+	log := &model.Log{ID: info.LogId}
+	log.Setup(s.env)
+	if err := log.Find(); err != nil {
+		return nil, errors.Wrapf(err, "problem finding log record for '%s'", info.LogId)
+	}
+
+	completedAt := time.Now()
+	if info.CompletedAt != nil {
+		var err error
+		completedAt, err = ptypes.Timestamp(info.CompletedAt)
+		if err != nil {
+			return nil, errors.Wrap(err, "problem converting completed_at timestamp")
+		}
+	}
+
+	log.Setup(s.env)
```

Comment:
This is pretty confusing, and I remember why we need to do it. Could we just make log.Find() or whatever set the env again?

Reply:
turns out we don't actually need to set the env twice, using the go driver's decode func doesn't wipe it out :)